logstash-core/lib/logstash/plugins/registry.rb

```diff
@@ -0,0 +1,83 @@
+# encoding: utf-8
+require 'singleton'
+require "rubygems/package"
+
+module LogStash
+  class Registry
+
+    ##
+    # Placeholder class for registered plugins
+    ##
+    class Plugin
```

Comment:
I don't think so, I like the separation of concerns here as it makes the different code path less polluted. If properly organized it could also benefit other parts of the stack with easier testing.